web/src/EducationBenchmarking.Web/Controllers/SchoolPlanningStepsController.cs

```diff
@@ -838,4 +839,68 @@ public async Task<IActionResult> PrimaryPupilFigures(string urn, int year)
             }
         }
     }
+
+    [HttpPost]
+    [Route("primary-pupil-figures")]
+    public async Task<IActionResult> PrimaryPupilFigures(string urn, int year,
+        [FromForm] PostPrimaryPupilFiguresViewModel formModel)
+    {
+        using (logger.BeginScope(new { urn, year }))
+        {
+            try
+            {
+                var school = await establishmentApi.GetSchool(urn).GetResultOrThrow<School>();
+                var plan = await benchmarkApi.GetFinancialPlan(urn, year).GetResultOrThrow<FinancialPlan>();
+                plan.PupilsNursery = formModel.PupilsNursery;
+                plan.PupilsMixedReceptionYear1 = formModel.PupilsMixedReceptionYear1Parsed;
+                plan.PupilsMixedYear1Year2 = formModel.PupilsMixedYear1Year2Parsed;
+                plan.PupilsMixedYear2Year3 = formModel.PupilsMixedYear2Year3Parsed;
+                plan.PupilsMixedYear3Year4 = formModel.PupilsMixedYear3Year4Parsed;
+                plan.PupilsMixedYear4Year5 = formModel.PupilsMixedYear4Year5Parsed;
+                plan.PupilsMixedYear5Year6 = formModel.PupilsMixedYear5Year6Parsed;
+                plan.PupilsReception = formModel.PupilsReceptionParsed;
+                plan.PupilsYear1 = formModel.PupilsYear1Parsed;
+                plan.PupilsYear2 = formModel.PupilsYear2Parsed;
+                plan.PupilsYear3 = formModel.PupilsYear3Parsed;
+                plan.PupilsYear4 = formModel.PupilsYear4Parsed;
+                plan.PupilsYear5 = formModel.PupilsYear5Parsed;
+                plan.PupilsYear6 = formModel.PupilsYear6Parsed;
+
+                ArgumentNullException.ThrowIfNull(plan.HasMixedAgeClasses, nameof(plan.HasMixedAgeClasses));
```

Comment:
Should we not put this before setting all the properties (fail fast)?

Reply:
To be honest, should use pattern matching on ternary below rather than guarding / throwing an exception.